bundles/org.openhab.binding.heliosventiremote/src/main/java/org/openhab/binding/heliosventiremote/internal/HeliosVentiRemoteHandler.java

```diff
@@ -0,0 +1,510 @@
+        variables.put((byte) 0xA9,
+                new HeliosVentiRemoteVariable(thing, HeliosVentiRemoteBindingConstants.CHANNEL_MIN_FANSPEED,
+                        (byte) 0xA9, true, HeliosVentiRemoteVariable.type.Fanspeed));
+        variables.put((byte) 0xA5,
+                new HeliosVentiRemoteVariable(thing, HeliosVentiRemoteBindingConstants.CHANNEL_MAX_FANSPEED,
+                        (byte) 0xA5, true, HeliosVentiRemoteVariable.type.Fanspeed));
+
+        variables.put((byte) 0x29,
+                new HeliosVentiRemoteVariable(thing, HeliosVentiRemoteBindingConstants.CHANNEL_FANSPEED, (byte) 0x29,
+                        true, HeliosVentiRemoteVariable.type.Fanspeed));
+
+        /*
+         * not yet supported
+         * variables.put((byte) 0xB1,
+         * new HeliosVentiRemoteVariable(thing, "DC_fan_outgoing", true, HeliosVentiRemoteVariable.type.Percent)); // in
+         *
+         * variables.put((byte) 0xB0,
+         * new HeliosVentiRemoteVariable(thing, "DC_fan_supply", true, HeliosVentiRemoteVariable.type.Percent)); // in %
+         *
+         * variables.put((byte) 0xAA, new HeliosVentiRemoteVariable(thing, "adjust_interval", true,
+         * HeliosVentiRemoteVariable.type.Number); // bit 0-4: adjust interval; bit 4: RH level setting: 1 = auto 0
+         * = manual; bit 5: 1 = boost, 0 = fireplace; bit 6: radiator type: 1 = water 0 = electric; bit 7: cascade on
+         *
+         * variables.put((byte) 0xA8,
+         * new HeliosVentiRemoteVariable(thing, "supply_stop_temp", (byte) 0xA8, true,
+         * HeliosVentiRemoteVariable.type.Temperature));
+         *
+         * variables.put((byte) 0xA7,
+         * new HeliosVentiRemoteVariable(thing, "preheat_temp", (byte) 0xA7, true,
+         * HeliosVentiRemoteVariable.type.Temperature));
+         *
+         * variables.put((byte) 0xA6,
+         * new HeliosVentiRemoteVariable(thing, "maintenance_interval", true, HeliosVentiRemoteVariable.type.Number));
+         * // in months
+         *
+         * variables.put((byte) 0xA4,
+         * new HeliosVentiRemoteVariable(thing, "set_temp", true, HeliosVentiRemoteVariable.type.Temperature));
+         */
+
+        /*
+         * variables.put((byte) 0xA3, new HeliosVentiRemoteVariable(thing, "state", (byte) 0xA3, true,
+         * HeliosVentiRemoteVariable.type.Number));
+         * This byte has a bit level interpretation and contains state and reset_maintenance
+         * variables.put((byte) 0xA3, "reset_maintenance");
+         *
+         * "power_state", 0xA3, bit 0
+         * "co2_state", 0xA3, bit 1
+         * "rh_state", 0xA3, bit 2
+         * "bypass_disabled", 0xA3, bit 3
+         */
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!isConnected()) {
+            connect(); // let's try to reconnect if the connection failed or was never established before
```

Comment:
the connect is probably quite fast, but an update of a value will take maybe 20ms due to several transmitted tx frames. Also the looping through the datapoints is not nice for runtime, especially when we add more of them in future. - I do not have a protocol spec and only took some addresses from a python script accessing the device and reverse-engineered what is interesting for me. So I also have not confidence that the timing of the frames is compliant.

Nevertheless it looks reasonable to implement some queue for the tx frames - which makes it also much simpler to satisfy more timing constraints (in case someone finds out some) and maybe also retry sending until we get an expected response (confirming the write) after setting a value.

The connect here is a hack to ensure that the early coming refresh are transmittable, because the startup of the serial connection takes some time and I first get some "No SerialPortProvider found for: /dev/ttyUSB0". - Do you know the serial transport in detail? Maybe it could be a dependency topic, that first the serial transport need to be initialized before this binding is started?

Reply:
No sorry, I'm not an expert there. But the queue is better for sure.

Reply:
I see history/dependencies.xml files in quite some bindings using the serial transport. Do I need this also or is it enough to put <feature>openhab-transport-serial</feature> into feature.xml to declare dependency of the heliosventialation binding on the serial transport?

Reply:
The files in `history/` are auto-generated from the `feature.xml` bythe maven-karaf-plugin. You don't need to add anything there.